tests/eosio.msig_tests.cpp

```diff
@@ -938,23 +1042,32 @@ BOOST_FIXTURE_TEST_CASE( approve_with_hash, eosio_msig_tester ) try {
                   ("level",         permission_level{ "alice"_n, config::active_name })
                   ("proposal_hash", trx_hash)
    );
-
-   transaction_trace_ptr trace;
-   control->applied_transaction.connect(
-   [&]( std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> p ) {
-      const auto& t = std::get<0>(p);
-      if( t->scheduled ) { trace = t; }
-   } );
-
-   push_action( "alice"_n, "exec"_n, mvo()
-                  ("proposer",      "alice")
-                  ("proposal_name", "first")
-                  ("executer",      "alice")
+   
+   transaction_trace_ptr trace = push_action( "alice"_n, "exec"_n, mvo()
+                                            ("proposer",      "alice")
+                                            ("proposal_name", "first")
+                                            ("executer",      "alice")
    );
 
    BOOST_REQUIRE( bool(trace) );
-   BOOST_REQUIRE_EQUAL( 1, trace->action_traces.size() );
    BOOST_REQUIRE_EQUAL( transaction_receipt::executed, trace->receipt->status );
+   BOOST_REQUIRE_EQUAL( 2, trace->action_traces.size() );
+
+   auto exec_trace = trace->action_traces.at(0);
+   BOOST_REQUIRE_EQUAL( "eosio.msig"_n, exec_trace.receiver );
+   BOOST_REQUIRE_EQUAL( "eosio.msig"_n, exec_trace.act.account );
+   BOOST_REQUIRE_EQUAL( "exec"_n, exec_trace.act.name );
+   BOOST_REQUIRE_EQUAL( 1, exec_trace.act.authorization.size() );
+   BOOST_REQUIRE_EQUAL( "alice"_n, exec_trace.act.authorization.at(0).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, exec_trace.act.authorization.at(0).permission );
+
+   auto reqauth_trace = trace->action_traces.at(1);
+   BOOST_REQUIRE_EQUAL( "eosio"_n, reqauth_trace.receiver );
+   BOOST_REQUIRE_EQUAL( "eosio"_n, reqauth_trace.act.account );
+   BOOST_REQUIRE_EQUAL( "reqauth"_n, reqauth_trace.act.name );
+   BOOST_REQUIRE_EQUAL( 1, reqauth_trace.act.authorization.size() );
+   BOOST_REQUIRE_EQUAL( "alice"_n, reqauth_trace.act.authorization.at(0).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, reqauth_trace.act.authorization.at(0).permission );
```

Comment:
Again lots of repetition of checks.

Perhaps it would be better to create a function which takes the original proposed transaction (e.g. `trx`) and the trace (e.g. `trace`) and does the checks on them to see if the trace properly executed the intended transaction. That way each test case can verify things executed properly without having a lot of redundant `BOOST_REQUIRE_EQUAL` statements.

Reply:
done